src/coredump/coredump.c

```diff
@@ -1321,12 +1339,12 @@ static int gather_pid_metadata_from_argv(
         /* We gather all metadata that were passed via argv[] into an array of iovecs that
          * we'll forward to the socket unit */
 
-        if (argc < _META_ARGV_MAX)
+        if (argc < _META_ARGV_MIN)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Not enough arguments passed by the kernel (%i, expected %i).",
                                        argc, _META_ARGV_MAX);
 
-        for (int i = 0; i < _META_ARGV_MAX; i++) {
+        for (int i = 0; i < MIN(argc, _META_ARGV_MAX); i++) {
                 _cleanup_free_ char *buf = NULL;
                 const char *t = argv[i];
 
```

Comment:
as mentioned above, plase skip this for the pidfd field, i.e. if meta_field_names[i] is null